src/app/components/mnemonic-input/mnemonic-word-input.tsx

```diff
@@ -0,0 +1,39 @@
+import { extractPhraseFromString } from '@app/common/utils';
+
+import { MnemonicInputField } from './mnemonic-input-field';
+
+interface MnemonicWordInputProps {
+  fieldNumber: number;
+  onUpdateWord(word: string): void;
+  onPasteEntireKey(word: string): void;
+}
+export function MnemonicWordInput({
+  fieldNumber,
+  onUpdateWord,
+  onPasteEntireKey,
+}: MnemonicWordInputProps) {
+  return (
+    <MnemonicInputField
+      name={`${fieldNumber}`}
+      autoCapitalize="off"
+      spellCheck={false}
+      data-testid={`mnemonic-input-${fieldNumber}`}
+      onPaste={e => {
+        const pasteValue = extractPhraseFromString(e.clipboardData.getData('text'));
+        if (pasteValue.includes(' ')) {
+          e.preventDefault();
+          //assume its a full key
+          // TODO should I do key validation here?
+          // maybe here is where to update all the name values also
+          onPasteEntireKey(pasteValue);
+        }
+      }}
+      onChange={(e: any) => {
+        e.preventDefault();
+        onUpdateWord(e.target.value);
+      }}
+      px="14px 16px"
```

Comment:
can we use spacing units here, even if requires two props?

Reply:
Sure, I'll update when I'm finalising everything, thanks!